lighthouse-core/scripts/legacy-javascript/create-polyfill-size-estimation.js

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * @param {number[]} arr
+ */
+function sum(arr) {
+  return arr.reduce((acc, cur) => acc + cur, 0);
+}
+
+function getPolyfillDependencies() {
+  /** @type {Map<string, string[]>} */
+  const polyfillDependencies = new Map();
+
+  for (const {name, coreJs3Module} of LegacyJavascript.getPolyfillData()) {
+    const folder = coreJs3Module.replace(/[^a-zA-Z0-9]+/g, '-');
+    const bundleMapPath =
+      `${VARIANT_DIR}/core-js-3-only-polyfill/${folder}/main.bundle.min.js.map`;
+    /** @type {LH.Artifacts.RawSourceMap} */
+    const bundleMap = JSON.parse(fs.readFileSync(bundleMapPath, 'utf-8'));
+    polyfillDependencies.set(name, bundleMap.sources.filter(s => s.startsWith('node_modules')));
+  }
+
+  const allPolyfillModules = [...polyfillDependencies.values()];
+  const commonModules = allPolyfillModules[0].filter(potentialCommonModule => {
+    return allPolyfillModules.every(modules => modules.includes(potentialCommonModule));
+  });
+  for (const [name, modules] of polyfillDependencies.entries()) {
+    polyfillDependencies.set(name, modules.filter(module => !commonModules.includes(module)));
+  }
+  polyfillDependencies.set('common', commonModules);
+
+  return polyfillDependencies;
+}
+
+async function main() {
+  const polyfillDependencies = getPolyfillDependencies();
+
+  const bundlePath =
+    `${VARIANT_DIR}/all-legacy-polyfills/all-legacy-polyfills-core-js-3/main.bundle.min.js`;
+  const bundleContents = fs.readFileSync(bundlePath, 'utf-8');
+  const bundleMap = JSON.parse(fs.readFileSync(bundlePath + '.map', 'utf-8'));
+  const artifacts = {
+    ScriptElements: [{requestId: '', src: '', content: bundleContents}],
+    SourceMaps: [{scriptUrl: '', map: bundleMap}],
+  };
+  // @ts-ignore
+  const bundles = await JsBundles.compute_(artifacts);
+  const bundleFileSizes = bundles[0].sizes.files;
+
+  const allModules = Object.keys(bundleFileSizes).filter(s => s.startsWith('node_modules'));
+  const moduleSizes = allModules.map(module => {
+    return bundleFileSizes[module];
+  });
+
+  /** @type {Map<string, number[]>} */
+  const polyfillDependenciesEncoded = new Map();
+  for (const [name, modules] of polyfillDependencies.entries()) {
+    polyfillDependenciesEncoded.set(name, modules.map(module => allModules.indexOf(module)));
+  }
+
+  const maxSize = sum(moduleSizes);
+  const baseSize = sum((polyfillDependencies.get('common') || []).map(m => bundleFileSizes[m]));
+  polyfillDependenciesEncoded.delete('common');
+  const polyfillDependencyGraphData = {
+    moduleSizes,
+    dependencies: [...polyfillDependenciesEncoded.entries()].reduce((acc, [name, modules]) => {
+      acc[name] = modules;
+      return acc;
+    }, /** @type {Record<string, number[]>} */ ({})),
+    maxSize,
+    baseSize,
+  };
+
+  const json = prettyJSONStringify(polyfillDependencyGraphData, {
+    tab: '  ',
+    spaceBeforeColon: '',
+    spaceInsideObject: '',
+    shouldExpand: _ => !Array.isArray(_),
```

Comment:
```suggestion
    shouldExpand: value => !Array.isArray(value),
```

we use `_` for unused variables, so this felt weird. don't really care what you call it other than that